### PR TITLE
Fixed typo errors

### PR DESCRIPTION
**Pull request by NamrataChangani**
## This PR is a:
- [ ] New topic
- [X] Content fix or rewrite
- [ ] Bug fix or improvement
 

## Summary
Fixed typo errors for `component's`

### CODE REVIEW

File: _data/toc/ui-library.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ pages:
     - label: Overview
       url: /ui-components/ui-component.html
 
-    - label: Using UI components' client-side
+    - label: Using UI component's client-side
```
Comment (shrielenee) [inline _data/toc/ui-library.yml]: Remove the apostrophe, this is simply plural - "components".

File: guides/v2.0/javascript-dev-guide/javascript/js_init.md [hunk 2]
```diff
@@ -45,7 +45,7 @@ When inserted in a certain element, the script is called only for this particula
 
 ##### How `data-mage-init` is processed {#init_process}
 
-On DOM ready, the `data-mage-init` attribute is parsed to extract components' names and configuration to be applied to the element.
+On DOM ready, the `data-mage-init` attribute is parsed to extract component's names and configuration to be applied to the element.
```
Comment (shrielenee) [inline guides/v2.0/javascript-dev-guide/javascript/js_init.md]: Revise to be "component".

File: guides/v2.0/ui-components/ui_components_js.md [hunk 3]
```diff
@@ -1,8 +1,8 @@
 ---
 group:  UI Library
 subgroup: A_Using_UI
-title: Using UI components' client-side
-menu_title: Using UI components' client-side
+title: Using UI component's client-side
```
Comment (shrielenee) [inline guides/v2.0/ui-components/ui_components_js.md]: Remove the apostrophe, this is simply plural - "components".

Comment (shrielenee) [reply guides/v2.0/ui-components/ui_components_js.md]: Same for below.

File: guides/v2.0/ui-components/ui_components_js.md [hunk 4]
```diff
@@ -14,13 +14,13 @@ This topic is aimed for developers, who need to reuse the [Magento UI Components
 
 The topic covers the following:
 
-- [UI components' configuration](#config)
+- [UI component's configuration](#config)
```
Comment (shrielenee) [inline guides/v2.0/ui-components/ui_components_js.md]: Revise to be "component".

File: guides/v2.0/ui-components/ui_components_js.md [hunk 5]
```diff
@@ -14,13 +14,13 @@ This topic is aimed for developers, who need to reuse the [Magento UI Components
...
 - [The most important UI component's properties](#main_properties)
 - [UI components’ properties used for linking](#comp_link)
 - [Description of the additional UI components](#comp_additional)
 - [JS UI components debugging](#comp_debug)
 
-## UI components' configuration {#config}
+## UI component's configuration {#config}
```
Comment (shrielenee) [inline guides/v2.0/ui-components/ui_components_js.md]: Revise to be "component".

File: guides/v2.0/ui-components/ui_components_js.md [hunk 6]
```diff
@@ -31,7 +31,7 @@ A UI component's behavior, configuration and structure is defined by the followi
 All these properties, options, and methods are available in the component template's scope.
 
 
-## Most important UI components' properties {#main_properties}
+## Most important UI component's properties {#main_properties}
```
Comment (shrielenee) [inline guides/v2.0/ui-components/ui_components_js.md]: Revise to be "component".

File: guides/v2.0/ui-components/ui_components_js.md [hunk 7]
```diff
@@ -76,7 +76,7 @@ This string is constructed from the following values:
  - `listing_top`: the value of the `name` attribute of the parent `<container name="listing_top">` component. 
  - `columns_controls`: the value of the `name` attribute of the component itself.
 
-## UI Components' properties used for linking {#comp_link}
+## UI Component's properties used for linking {#comp_link}
```
Comment (shrielenee) [inline guides/v2.0/ui-components/ui_components_js.md]: Revise to be "component".

File: guides/v2.1/frontend-dev-guide/translations/translate_theory.md [hunk 8]
```diff
@@ -42,7 +42,7 @@ To make sure the strings of an email template are added to the dictionary, use t
 
 Custom email templates <a href="{{ page.baseurl }}/frontend-dev-guide/templates/template-email.html#customize-email-admin" target="_blank">added using the Admin panel</a>, are not stored in the file system, and their stings are not added to the dictionary.
 
-<h2 id="add_strings_ui_html">Strings added in UI components' templates</h2>
+<h2 id="add_strings_ui_html">Strings added in UI component's templates</h2>
```
Comment (shrielenee) [inline guides/v2.1/frontend-dev-guide/translations/translate_theory.md]: Revise to be "component".

File: guides/v2.1/frontend-dev-guide/translations/translate_theory.md [hunk 9]
```diff
@@ -54,7 +54,7 @@ To ensure that the text you add in `.html` templates of UI components is added t
 
 - when a string is added with no binding to an HTML element:
 
-{% highlight HTML%}
+{% highlight HTML%}:q
```
Comment (shrielenee) [inline guides/v2.1/frontend-dev-guide/translations/translate_theory.md]: Remove the ":q".

Change "when" to "When" in these bullet points.

File: guides/v2.1/javascript-dev-guide/javascript/js_init.md [hunk 10]
```diff
@@ -45,7 +45,7 @@ When inserted in a certain element, the script is called only for this particula
 
 ##### How `data-mage-init` is processed {#init_process}
 
-On DOM ready, the `data-mage-init` attribute is parsed to extract components' names and configuration to be applied to the element.
+On DOM ready, the `data-mage-init` attribute is parsed to extract component's names and configuration to be applied to the element.
```
Comment (shrielenee) [inline guides/v2.1/javascript-dev-guide/javascript/js_init.md]: Revise to be "component".

File: guides/v2.1/ui_comp_guide/bk-ui_comps.md [hunk 11]
```diff
@@ -104,7 +104,7 @@ A UI component can be bound to one or more {% glossarytooltip a2aff425-07dd-4bd6
 
 A particular instance of a UI component is defined primarily by the following:
 
-1. [definition.xml]({{ site.mage2100url }}app/code/Magento/Ui/view/base/ui_component/etc/definition.xml): default components' configuration. Can be extended in custom modules.
+1. [definition.xml]({{ site.mage2100url }}app/code/Magento/Ui/view/base/ui_component/etc/definition.xml): default component's configuration. Can be extended in custom modules.
```
Comment (shrielenee) [inline guides/v2.1/ui_comp_guide/bk-ui_comps.md]: Revise to be "component".

File: guides/v2.1/ui_comp_guide/concepts/ui_comp_uiclass_concept.md [hunk 12]
```diff
@@ -10,7 +10,7 @@ github_link: ui_comp_guide/concepts/ui_comp_uiclass_concept.md
 
 ## What is `uiClass`?
 
-The `uiClass` is an abstract class from which all components are extended. The `uiClass` is a low-level class and is rarely used as direct parent for UI components' classes.
+The `uiClass` is an abstract class from which all components are extended. The `uiClass` is a low-level class and is rarely used as direct parent for UI component's classes.
```
Comment (shrielenee) [inline guides/v2.1/ui_comp_guide/concepts/ui_comp_uiclass_concept.md]: Revise to be "component".

File: guides/v2.1/ui_comp_guide/concepts/ui_comp_uielement_concept.md [hunk 13]
```diff
@@ -64,7 +64,7 @@ When creating a new component, use the `uiElement` class as a direct parent, if
...
           this.trackable;     // getter
 
-- The `initModules()` method initializes external UI components' instances and links them to local variables. This method works with the `modules` section of the configuration, where we define the name of the external instance and the local variable. The `modules` section is an object where each key is the variable's name and each value is the instance's name. A developer does not need to be concerned with the instantiation of the module (external instance), because the `initModules()` method will resolve the variable when the instance is instantiated.
+- The `initModules()` method initializes external UI component's instances and links them to local variables. This method works with the `modules` section of the configuration, where we define the name of the external instance and the local variable. The `modules` section is an object where each key is the variable's name and each value is the instance's name. A developer does not need to be concerned with the instantiation of the module (external instance), because the `initModules()` method will resolve the variable when the instance is instantiated.
```
Comment (shrielenee) [inline guides/v2.1/ui_comp_guide/concepts/ui_comp_uielement_concept.md]: Revise to be "component".

File: guides/v2.1/ui_comp_guide/concepts/ui_comp_xmldeclaration_concept.md [hunk 14]
```diff
@@ -15,7 +15,7 @@ This topic discusses the {% glossarytooltip 8c0645c5-aa6b-4a52-8266-5659a8b9d079
...
+In a typical Magento `.xml` layout file we see a `<head/>` node, `<title/>` node with the name of the page, and sometimes [links to CSS and JS files]({{ page.baseurl }}/frontend-dev-guide/layouts/xml-manage.html#layout_markup_css). There are other nodes as well, the most important for us now is the [`<referenceContainer/>` node]({{ page.baseurl }}/frontend-dev-guide/layouts/xml-instructions.html#fedg_layout_xml-instruc_ex_ref). (The `name` attribute in this node is responsible for the position of the container on the page.). [Basic]({{ page.baseurl }}/ui_comp_guide/bk-ui_comps.html#general-structure) UI components are declared in this node. All nested components are declared in the basic component's instances configuration files (not in the page layouts).
```
Comment (shrielenee) [inline guides/v2.1/ui_comp_guide/concepts/ui_comp_xmldeclaration_concept.md]: Revise to be "component".

File: guides/v2.2/ui_comp_guide/bk-ui_comps.md [hunk 15]
```diff
@@ -104,7 +104,7 @@ A UI component can be bound to one or more {% glossarytooltip a2aff425-07dd-4bd6
 
 A particular instance of a UI component is defined primarily by the following:
 
-1. [definition.xml](https://github.com/magento/magento2/blob/2.2/app/code/Magento/Ui/view/base/ui_component/etc/definition.xml): default components' configuration. Can be extended in custom modules.
+1. [definition.xml](https://github.com/magento/magento2/blob/2.2/app/code/Magento/Ui/view/base/ui_component/etc/definition.xml): default component's configuration. Can be extended in custom modules.
```
Comment (shrielenee) [inline guides/v2.2/ui_comp_guide/bk-ui_comps.md]: Revise to be "component".

File: guides/v2.2/ui_comp_guide/components/basic-attributes.md [hunk 16]
```diff
@@ -50,7 +50,7 @@ The following attributes are available for all UI components, including the basi
 </table>
 
 
-## Basic components' options
+## Basic component's options
```
Comment (shrielenee) [inline guides/v2.2/ui_comp_guide/components/basic-attributes.md]: Revise to be "Basic component options".

File: guides/v2.2/ui_comp_guide/concepts/ui_comp_xmlconfig_structure.md [hunk 17]
```diff
@@ -10,7 +10,7 @@ github_link: ui_comp_guide/concepts/ui_comp_xmlconfig_structure.md
 
 ## Overview
 
-This topic describes the basic elements used in XML configuration files for declaring  UI components. These elements co structure is strict and third party developers must follow it when customizing existing UI components' configuration or declaring new ones.  
+This topic describes the basic elements used in XML configuration files for declaring  UI components. These elements co structure is strict and third party developers must follow it when customizing existing UI component's configuration or declaring new ones.
```
Comment (shrielenee) [inline guides/v2.2/ui_comp_guide/concepts/ui_comp_xmlconfig_structure.md]: Change "co structure" to "core structure".

File: guides/v2.3/ui_comp_guide/bk-ui_comps.md [hunk 18]
```diff
@@ -104,7 +104,7 @@ A UI component can be bound to one or more {% glossarytooltip a2aff425-07dd-4bd6
 
 A particular instance of a UI component is defined primarily by the following:
 
-1. `<Magento_Ui_module_dir>/view/base/ui_component/etc/definition.xml`: default components' configuration. Can be extended in custom modules.
+1. `<Magento_Ui_module_dir>/view/base/ui_component/etc/definition.xml`: default component's configuration. Can be extended in custom modules.
```
Comment (shrielenee) [inline guides/v2.3/ui_comp_guide/bk-ui_comps.md]: Revise to be "component".

**Comment by NamrataChangani** [comment]
@shrielenee I have fixed all mentioned changes. Please verify at once.

File: guides/v2.0/javascript-dev-guide/javascript/js_init.md [hunk 19]
```diff
@@ -45,7 +45,7 @@ When inserted in a certain element, the script is called only for this particula
 
 ##### How `data-mage-init` is processed {#init_process}
 
-On DOM ready, the `data-mage-init` attribute is parsed to extract components' names and configuration to be applied to the element.
+On DOM ready, the `data-mage-init` attribute is parsed to extract components names and configuration to be applied to the element.
```
Comment (shrielenee) [inline guides/v2.0/javascript-dev-guide/javascript/js_init.md]: "component" not "components"

File: guides/v2.0/ui-components/ui_components_js.md [hunk 20]
```diff
@@ -14,13 +14,13 @@ This topic is aimed for developers, who need to reuse the [Magento UI Components
 
 The topic covers the following:
 
-- [UI components' configuration](#config)
+- [UI components configuration](#config)
```
Comment (shrielenee) [inline guides/v2.0/ui-components/ui_components_js.md]: "component" not "components"

File: guides/v2.0/ui-components/ui_components_js.md [hunk 21]
```diff
@@ -14,13 +14,13 @@ This topic is aimed for developers, who need to reuse the [Magento UI Components
...
 - [The most important UI component's properties](#main_properties)
 - [UI components’ properties used for linking](#comp_link)
 - [Description of the additional UI components](#comp_additional)
 - [JS UI components debugging](#comp_debug)
 
-## UI components' configuration {#config}
+## UI components configuration {#config}
```
Comment (shrielenee) [inline guides/v2.0/ui-components/ui_components_js.md]: "component" not "components"

**Pull request merged**